src/main/java/com/st/utopia/cancellationservice/model/Ticket.java

```diff
@@ -0,0 +1,163 @@
+import javax.persistence.Entity;
+import javax.persistence.Table;
+
+/**
+ * A Ticket model.
+ *
+ * 
+ * Model used for the purpose of the cancellation microservice.
+ * 
+ * There are no set methods for information about the ticket such as flight
+ * number, row/seat, or class.
+ * 
+ * Upon cancelling the ticket, the microservice needs to set the following
+ * fields as null: -reserver -price -reservation_timeout -booking_id
+ * 
+ * This model does not need to define a composite key since the cancellation
+ * microservice will search for the ticket via booking_id which is unique
+ * 
+ * 
+ * @author Al-amine AHMED MOUSSA
+ */
+
+@Entity
+@Table(name = "tbl_tickets")
+public class Ticket {
+	/**
+	 * The three fields that together constitute the identity of a Ticket.
+	 */
+	@EmbeddedId
+	private TicketIdentity id;
+
+	@Column(name = "class")
+	private Integer tier; // class is a keyword
+
+	private Integer reserver;
+
+	private Integer price;
+
+	private LocalDateTime reservationTimeout;
+
+	private String bookingId;
+
+	/**
+	 * the Getters and Setters of the Class
+	 */
+	public TicketIdentity getId() {
+		return id;
+	}
+
+	public Integer getFlight() {
+		if (id == null) {
+			return null;
+		} else {
+			return id.getFlight();
+		}
+	}
+
+	public Integer getRow() {
+		if (id == null) {
+			return null;
+		} else {
+			return id.getRow();
+		}
+	}
+
+	public char getSeat() {
+		if (id == null) {
+			return 0;
+		} else {
+			return id.getSeat();
+		}
+	}
+
+	public Integer getTier() {
+		return tier;
+	}
+
+	public void setTier(Integer tier) {
+		this.tier = tier;
+	}
+
+	public Integer getReserver() {
+		return reserver;
+	}
+
+	public void setReserver(Integer reserver) {
+		this.reserver = reserver;
+	}
+
+	public Integer getPrice() {
+		return price;
+	}
+
+	public void setPrice(Integer price) {
+		this.price = price;
+	}
+
+	public LocalDateTime getReservationtimeout() {
+		return reservationTimeout;
+	}
+
+	public void setReservationtimeout(LocalDateTime reservationTimeout) {
+		this.reservationTimeout = reservationTimeout;
+	}
+
+	public String getBookingid() {
+		return bookingId;
+	}
+
+	public void setBookingid(String bookingId) {
+		this.bookingId = bookingId;
+	}
+
+	/*
+	 * we need empty Constructor for Hibernate Jpa Purposes
+	 */
+	public Ticket() {
+	}
+
+	public Ticket(TicketIdentity id) {
+		this.id = id;
+	}
+
+	public Ticket(TicketIdentity id, Integer tier, Integer reserver, Integer price, LocalDateTime reservationTimeout,
+			String bookingId) {
+		this.id = id;
+		this.tier = tier;
+		this.reserver = reserver;
+		this.price = price;
+		this.reservationTimeout = reservationTimeout;
+		this.bookingId = bookingId;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(id);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		} else if (obj instanceof Ticket) {
+			return Objects.equals(id, ((Ticket) obj).id);
+		}
+		Ticket other = (Ticket) obj;
```

Comment:
By this line we *know* that `obj` is *not* a `Ticket`, so this will cause a `ClassCastException`. Instead, just return `false`. (I would put it in an `else` block, but that's a matter of stylistic preference.)